mono/mini/debugger-engine.c

```diff
@@ -1610,19 +1610,20 @@ mono_debugger_free_objref (gpointer value)
 MonoClass *
 get_class_to_get_builder_field(DbgEngineStackFrame *frame)
 {
+	StackFrame *the_frame = (StackFrame *)frame;
 	ERROR_DECL (error);
 	gpointer this_addr = get_this_addr (frame);
 	MonoClass *original_class = frame->method->klass;
 	MonoClass *ret;
-	if (!m_class_is_valuetype (original_class) && mono_class_is_open_constructed_type (m_class_get_byval_arg (original_class))) {
+	if (mono_class_is_open_constructed_type (m_class_get_byval_arg (original_class))) {
 		MonoObject *this_obj = *(MonoObject**)this_addr;
 		MonoGenericContext context;
 		MonoType *inflated_type;
 
 		if (!this_obj)
 			return NULL;
 			
-		context = mono_get_generic_context_from_stack_frame (frame->ji, this_obj->vtable);
```

Comment:
Looking at the rest of the PR, I think the change here is the important one. probably passing `this_obj->vtable` resulted in a `context` that doesn't actually have any generic instance in it, so `inflated_type`, below` is just the original class with the gsharing `T_REF` in it.

@vargaz what do you think?